### ISSUE TITLE
Add a `@Recurse`-like annotation for CLI parsing

### ISSUE THREAD

**Issue by jvican** [olafurpg/metaconfig#40]
Let's say I have:
```scala
case class Opts(workingDirectory: String = System.getProperty("user.dir"))
case class OptsWrapper(opts: Opts = Opts())

implicit val optsSurface = deriveSurface[Opts]
implicit val optsDecoder = deriveDecoder[Opts](Opts())

implicit val optsWSurface = deriveSurface[OptsWrapper]
implicit val optsWDecoder = deriveDecoder[OptsWrapper](OptsWrapper())
```
  
When I parse that as CLI args, `--workingDirectory` doesn't work, but `--opts.workingDirectory` does. I'd like to find a way to make the latter work.
  
For that, I propose an annotation for the `opts` field in `OptsWrapper` that tells metaconfig to allow dropping the `opts` prefix *when parsing from CLI args*, in a similar fashion to how `@Recurse` works in caseapp. Metaconfig should check at compile-time that there are no conflicting names between the fields defined in `Opts` and the fields defined in `OptsWrapper`.
  
The benefit of doing this is that the users of the CLI don't need to know about how I internally structure my code. I could inline manually the fields of `Opts`, but as the case class grows, that becomes a maintenance burden and gives a poor UX when using the API (no automatic `copy(opts = otherOpts)` for every command in which I depend on `opts`).
  
A better name than `@Recurse` would be `@Inline`.

**Comment by olafurpg** [comment]
This is a good idea! I have a feeling it should not be too difficult to support. I still haven't gotten around to polish the CLI parser but I hope to get around to it sometime this year, will keep this ticket open for then :)

**Comment by olafurpg** [comment]
I ended up needing this feature myself :) Fixed in #51 (commit https://github.com/olafurpg/metaconfig/pull/51/commits/1837b24981908e8572a72b8d11a8b29a2efa0aab) and documented in #52 

Docs are already live on the website https://olafurpg.github.io/metaconfig/#inline
